src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -2291,6 +2291,18 @@ public static SyntaxKind GetPredefinedKeywordKind(SpecialType specialType)
 
         public static SimpleNameSyntax GetRightmostName(this ExpressionSyntax node)
         {
+            var awaitExpression = node as AwaitExpressionSyntax;
+            if (awaitExpression != null && awaitExpression.Expression != null)
+            {
+                return awaitExpression.Expression.GetRightmostName();
+            }
+
+            var invocationExpression = node as InvocationExpressionSyntax;
+            if (invocationExpression != null && invocationExpression.Expression != null)
```

Comment:
This is especially concerning.  An invocation doesn't even have a rightmost name.  Because it's got more stuff to the rigth.